tools/python/migration-scripts/Ingest_S3_to_postgressql/README.md

```diff
@@ -0,0 +1,115 @@
+# PostgreSQL CSV Ingestion Tool
+
+## Overview
+
+This tool provides a high-performance solution for loading CSV files into PostgreSQL databases. It leverages multi-threading and connection pooling to process multiple files in parallel, significantly reducing data loading time.
+
+## Features
+
+- **Multi-threaded Processing**: Load multiple CSV files simultaneously
+- **Connection Pooling**: Efficiently manage database connections
+- **Automatic Column Detection**: Dynamically extract column names from CSV headers
+- **Retry Logic**: Handle transient errors with exponential backoff
+- **File Management**: Moves processed files to a designated directory so retrying is resuming but not restarting
+- **Comprehensive Logging**: Detailed logs for monitoring and troubleshooting
+- **Error Notifications**: Optional SNS notifications for failures
+- **Secure Credentials**: Retrieve database passwords from AWS Secrets Manager
+
+## Prerequisites
+
+- Python 3.6+
+- Aurora/RDS Postgres Cluster schema, database and table with correct datatypes created.
+- AWS credentials configured if using Secrets Manager and Simple Notification Services
+- Download CSV files generated by Timestream unload tool
+
+
+## Installation
+
+```bash
+pip install boto3 psycopg2-binary
+```
+
+## Usage
+
+```bash
+python copy_postgres.py \
+  --table TABLE_NAME \
+  --database DATABASE_NAME \
+  --csv_files_dir "/path/to/csv/files/*.csv" \
+  --host DATABASE_HOST \
+  --secret_name SECRET_NAME \
+  [--schema SCHEMA] \
+  [--user USERNAME] \
+  [--port PORT] \
+  [--parallel_threads THREADS] \
+  [--processed_dir PROCESSED_DIR] \
+  [--logs_dir LOGS_DIR] \
+  [--sns_topic_arn SNS_TOPIC_ARN]
+```
+
+## Parameters
+
+### Required Parameters
+
+- `--table`: Target PostgreSQL table name
```

Comment:
Why not support table creation

Reply:
Discussed and table creation will not be supported due to indexes and table partition by time.